mlflow/data/spark_dataset.py

```diff
@@ -36,16 +36,24 @@ def __init__(
         targets: Optional[str] = None,
         name: Optional[str] = None,
         digest: Optional[str] = None,
+        predictions: Optional[str] = None,
```

Comment:
While we're here, support passing `predictions` with spark dataframes, else Databricks evaluation use cases might fail if users pass in Spark dataframes *and* decide to manually specify prediction column names (instead of letting Databricks evaluation infer the column name)